### PR TITLE
Misc e2e test fixes

### LINKED ISSUE #44059: [Flaky] shopper/order-email-receiving.spec.js
should receive order email after purchasing an item 

Failed on: https://github.com/woocommerce/woocommerce/actions/runs/7649123646

### LINKED ISSUE #44060: [Flaky] merchant/products/block-editor/disable-block-product-editor.spec.js
can be disabled from the header 

Failed on: https://github.com/woocommerce/woocommerce/actions/runs/7649123646

### LINKED ISSUE #44172: [Flaky] shopper/cart-checkout-block-calculate-tax.spec.js
that inclusive tax is displayed properly in blockbased Cart & Checkout pages 

Failed on: https://github.com/woocommerce/woocommerce/actions/runs/7708338662

### LINKED ISSUE #44252: [Flaky] shopper/cart-checkout-restricted-coupons.spec.js
coupon can be used by the right customer (email restricted) but only once 

Failed on: https://github.com/woocommerce/woocommerce/actions/runs/7735841631

### PR DESCRIPTION
**Pull request by lanej0**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

There's a number of tests that have become flaky recently.  Here's some fixes.

Closes #44060 
Closes #44059 
Closes #44172 
Closes #44252 

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Make sure CI passes

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce/tests/e2e-pw/playwright.config.js [hunk 1]
```diff
@@ -22,7 +22,7 @@ const config = {
 	testDir: 'tests',
 	retries: CI ? 2 : 0,
 	repeatEach: REPEAT_EACH ? Number( REPEAT_EACH ) : 1,
-	workers: CI ? 1 : 4,
+	workers: 1,
```
Comment (adimoldovan) [inline plugins/woocommerce/tests/e2e-pw/playwright.config.js:25]: Was this change intentional or maybe leftover from you testing? I would be ok with it either way, I usually disable it locally.

**Review by adimoldovan — APPROVED** [review]
The changes look good to me, I only left one comment for the workers update as I'm not sure you meant to push that change.

File: plugins/woocommerce/tests/e2e-pw/playwright.config.js (hunk 1 above)
Comment (lanej0) [reply plugins/woocommerce/tests/e2e-pw/playwright.config.js:25]: It was intentional because on CI, we're running tests 1 at a time.  We used to be working towards 4 at a time everywhere, but since switching directions, it doesn't make sense to still run tests in parallel locally -- there's just too many failures.
